### PR TITLE
bump react rules

### PR DESCRIPTION
**Pull request by aju**
https://github.com/yannickcr/eslint-plugin-react

### CODE REVIEW

File: packages/eslint-config-brainly-react/rules/react.js [hunk 1]
```diff
@@ -14,12 +14,18 @@ module.exports = {
...
     'react/boolean-prop-naming': 'off',
+    // Forbid "button" element without an explicit "type" attribute
+    'react/button-has-type:': 'off',
     // Prevent extraneous defaultProps on components
     'react/default-props-match-prop-types': 'error',
+    // Rule enforces consistent usage of destructuring assignment in component
+    'react/destructuring-assignment': 'off',
     // Prevent missing displayName in a React component definition
-    'react/display-name': 'error',
+    'react/display-name': 'off',
```
Comment (aju) [inline packages/eslint-config-brainly-react/rules/react.js]: was not helpfull

File: packages/eslint-config-brainly-react/rules/react.js [hunk 2]
```diff
@@ -49,7 +55,7 @@ module.exports = {
     // Prevent usage of isMounted
     'react/no-is-mounted': 'error',
     // Prevent multiple component definition per file
-    'react/no-multi-comp': 'error',
+    'react/no-multi-comp': 'off',
```
Comment (aju) [inline packages/eslint-config-brainly-react/rules/react.js]: was not helpfull

File: packages/eslint-config-brainly-react/rules/react.js [hunk 3]
```diff
@@ -60,10 +66,14 @@ module.exports = {
...
     // Prevent using string references in ref attribute.
     'react/no-string-refs': 'error',
+    // Prevent this from being used in stateless functional components
+    'react/no-this-in-sfc': 'error',
     // Prevent invalid characters from appearing in markup
     'react/no-unescaped-entities': 'error',
     // Prevent usage of unknown DOM property (fixable)
     'react/no-unknown-property': 'error',
+    // Prevent usage of unsafe lifecycle methods
+    'react/no-unsafe': 'error',
```
Comment (aju) [inline packages/eslint-config-brainly-react/rules/react.js]: We will have to add ignore for some files right now

File: packages/eslint-config-brainly-react/package.json [hunk 4]
```diff
@@ -20,13 +20,6 @@
     "eslint-config-brainly": "^2.1.0"
   },
   "peerDependencies": {
-    "eslint": "^4.5.0",
```
Comment (matikrk) [inline packages/eslint-config-brainly-react/package.json]: why deleting peerDependencies? It allows easy install everything in project

Comment (aju) [reply packages/eslint-config-brainly-react/package.json]: we have dependency on "eslint-config-brainly" it has it all as "peerDependencies" - we keep it in sync in one place :)

Comment (matikrk) [reply packages/eslint-config-brainly-react/package.json]: But command from readme will work right now?  `npx install-peerdeps --dev eslint-config-brainly-react`

File: packages/eslint-config-brainly-react/package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-config-brainly-react",
-  "version": "2.1.1",
+  "version": "2.2.0",
```
Comment (matikrk) [inline packages/eslint-config-brainly-react/package.json]: 👍  After merge remember to push to npm

**Pull request merged**